apollo-federation/src/query_plan/query_planning_traversal.rs

```diff
@@ -472,29 +472,7 @@ impl<'a: 'b, 'b> QueryPlanningTraversal<'a, 'b> {
         //   later condition is that `selection` is originally a supergraph selection, but that we're looking to apply "as-is" to a subgraph.
         //   But suppose it has a `... on I` where `I` is an interface. Then it's possible that `I` includes "more" types in the supergraph
         //   than in the subgraph, and so we might have to type-explode it. If so, we cannot use the selection "as-is".
-        let mut has_inconsistent_abstract_types: Option<bool> = None;
```

Comment:
We could replace this with `let mut has_inconsistent_abstract_types: OnceCell<bool> = OnceCell::new()`, and then execute the condition with `has_inconsistent_abstract_types.get_or_init()`, which is exactly equivalent to what the closure code is doing. (and imo a little better than having a separate bool to track whether the code has been executed)